### ISSUE TITLE
Awaitable pipeline commands

### ISSUE THREAD

**Issue by ArtemIsmagilov** [aembke/fred.rs#299]
Fred version - 9.3.0
Redis version - 7.0.0
Platform - linux
Deployment type - cluster|sentinel|centralized

**Describe the bug**
Hi. Please correct me where I'm wrong.

It looks like your pipeline implementation is not quite correct.

In short, for each command you somehow make a request to the server and do await. However, after looking at the documentation and the source code, for example Python, I saw that the pipeline is implemented without await waiting for a response from the server. It looks like sending to the server should be done once, and the commands from the client should be stored in a vector. I'm new to Rust and could have missed something. Explain these details as a newbie, I would be very grateful.
**To Reproduce**
0) Redis doc pipeline https://redis.io/docs/latest/develop/use/pipelining/
1) Python client pipeline example https://github.com/redis/redis-py/blob/d6ddb0d66bc4bd28c99d9cf148c9085117cc8ce9/redis/asyncio/cluster.py#L1375
2) Your implementation https://github.com/aembke/fred.rs/blob/2bc4c9ec58b0492fc4eee56ce36ea2452151a371/examples/pipeline.rs#L15

**Logs**
(If possible set `RUST_LOG=fred=trace` and run with `--features debug-ids`)

**Comment by aembke** [comment]
Hi @ArtemIsmagilov, 

No problem, I can see the confusion, but the `await` on each of the commands is just there so I could reuse the same interfaces with the other client structs, and therefore callers could make their code generic over pipelined and non-pipelined sequences of commands.

For what it's worth, it does work how you'd expect under the hood. When you call a command on a pipeline it queues it in memory and the commands aren't send until you call `all`, `last`, or `try_all`, and those intermediate `await` points return immediately without actually waiting on anything.

**Comment by ArtemIsmagilov** [comment]
Hi @aembke, thanks for helping me clarify this. It seems like the only difference is in the user experience. The most important thing is, as you said, the pipeline works exactly as intended without any real wait from the server. Great!

**Comment by aembke** [comment]
No problem, feel free to open another issue if you have any other questions/issues.
